### PR TITLE
update the parser to support new play store layout

### PR DESCRIPTION
**Pull request by humanolaranja**
#179

### CODE REVIEW

File: lib/src/play_store_search_api.dart [hunk 1]
```diff
@@ -138,11 +138,42 @@ class PlayStoreResults {
   static String? version(Document response) {
     String? version;
     try {
-      final additionalInfoElements = response.getElementsByClassName('hAyfc');
-      final versionElement = additionalInfoElements.firstWhere(
-        (elm) => elm.querySelector('.BgcNfc')!.text == 'Current Version',
-      );
-      final storeVersion = versionElement.querySelector('.htlgb')!.text;
+      const patternName = ",\"name\":\"";
```
Comment (larryaasen) [inline lib/src/play_store_search_api.dart:141]: @humanolaranja Thanks for the code changes in this PR. I would like to maintain support for the old Play Store format, which will also keep the existing tests passing. Can you call this new code in a new method only when the current code fails? Then, you could add a new test_play_store_page for testing.

Comment (humanolaranja) [reply lib/src/play_store_search_api.dart:141]: Sure, I'll update it

File: lib/src/play_store_search_api.dart [hunk 2]
```diff
@@ -107,33 +107,55 @@ class PlayStoreResults {
...
 
       return releaseNotes;
     } catch (e) {
-      print('upgrader: PlayStoreResults.releaseNotes exception: $e');
+      print('upgrader: PlayStoreResults.redesignedReleaseNotes exception: $e');
     }
     return null;
   }
 
+  static String? multilineDescription(
```
Comment (larryaasen) [inline lib/src/play_store_search_api.dart:142]: @humanolaranja There is already a field named description and this new method `multilineDescription` is for release notes, so maybe change the name to `multilineReleaseNotes`?

File: test/play_store_test.dart [hunk 3]
```diff
@@ -51,6 +51,22 @@ void main() {
...
+    final playStore = PlayStoreSearchAPI();
+    playStore.client = client;
+
+    final response = await playStore.lookupById('com.testing.test4');
+    expect(response, isNotNull);
+    expect(response, isInstanceOf<Document>());
+
+    expect(PlayStoreResults.releaseNotes(response!),
+        'Minor updates and improvements.');
+    expect(PlayStoreResults.version(response), '2.3.0');
```
[the remark this answers is not in the document]
Comment (humanolaranja) [reply test/play_store_test.dart:65]: Sure, just finishing this to remove the WIP flag =D

**Review by larryaasen — APPROVED** [review]

**Comment by larryaasen** [comment]
I just published version [4.2.1-alpha.1](https://pub.dev/packages/upgrader/versions/4.2.1-alpha.1) on pub.dev with this update. Please test out this pre-release version ASAP and provide feedback. After it has been determined that this solution works well, I will release a normal version.

**Pull request merged**